File: Leetcode/617_Merge_Two_Binary_Trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* mergeTrees(TreeNode* t1, TreeNode* t2) {
        if(t1 == nullptr && t2 == nullptr)
        {
            return nullptr;
        }
        int val = 0;
        if(t1 == nullptr)
        {
            val = t2->val;
        }
        else if( t2 == nullptr)
        {
            val = t1->val;    
        }
        else
        {
            val = t1->val + t2->val;
        }
        TreeNode* t = new TreeNode(val);
        
        t->left = mergeTrees(t1 == nullptr ? nullptr : t1->left, t2 == nullptr ? nullptr : t2->left);
        t->right = mergeTrees(t1 == nullptr ? nullptr : t1->right, t2 == nullptr ? nullptr : t2->right);
        
        return t;
    }
    
};

File: Leetcode/0617/617_Merge_Two_Binary_Trees.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} t1
 * @param {TreeNode} t2
 * @return {TreeNode}
 */
var mergeTrees = function (t1, t2) {
    if (t1 || t2) {
        const t1Val = t1 ? t1.val : 0;
        const t2Val = t2 ? t2.val : 0;

        const root = new TreeNode(t1Val + t2Val);

        root.left = mergeTrees(t1 ? t1.left : null, t2 ? t2.left : null);
        root.right = mergeTrees(t1 ? t1.right : null, t2 ? t2.right : null);

        return root;
    }
    return null;
};